beamer/agent.py

```diff
@@ -79,44 +79,37 @@ def __init__(self, config: Config):
         )
         self._event_processor = EventProcessor(self.context)
 
-        self._contract_monitor_l2a = ContractEventMonitor(
-            "RequestManager",
-            request_manager,
-            l2a_contracts_info["RequestManager"].deployment_block,
-            self._event_processor.add_events,
-            self._event_processor.mark_sync_done,
+        self._contract_monitor_l2a = EventMonitor(
```

Comment:
Nitpick: consider renaming the instance variable to `_event_monitor_l2a` to be consistent with the new type name.